tls/s2n_handshake.h

```diff
@@ -184,6 +196,8 @@ struct s2n_handshake {
 
     /* Indicates that this is a renegotiation handshake */
     unsigned renegotiation : 1;
+
+    s2n_state_machine state_machine;
```

Comment:
Should the current state machine persist after s2n_renegotiate_wipe? Or should it manually be set back to S2N_STATE_MACHINE_TLS12 when renegotiating? Not sure if this is an issue or not.

Reply:
That is a good question. 
Renegotiates are only valid for < TLS13 handshakes. So s2n_renegotiate_wipe() would wipe the connection struct, setting state_machine back to zero(S2N_STATE_MACHINE_INITIAL). And S2N_STATE_MACHINE_INITIAL will reference the TLS12 state machine because thats the state machine we start out on. 
So I think we should be good in this case. It would be more of an issue if renegotiates were valid for TLS13, because then we might want to actual preserve the state machine.

Reply:
Right, that makes sense. I guess I was just thinking that since it gets set to initial, it has the possibility of then getting set to TLS 1.3, when it would have been 1.2 before. But since it's an entirely new handshake, I suppose it wouldn't actually be changing state machines, so I guess this wouldn't be a problem from the state machine's perspective.